lib/backend/watchersyncer/watchercache.go

```diff
@@ -117,30 +117,15 @@ mainLoop:
 				kvp.Value = nil
 				wc.handleWatchListEvent(kvp)
 			case api.WatchError:
-				// Handle a WatchError.  First determine if the error type indicates that the
-				// watch has closed, and if so we'll need to resync and create a new watcher.
-				wc.results <- event.Error
-
-				if e, ok := event.Error.(cerrors.ErrorWatchTerminated); ok {
-					wc.logger.Debug("Received watch terminated error - recreate watcher")
-					if !e.ClosedByRemote {
-						// If the watcher was not closed by remote, reset the currentWatchRevision.  This will
-						// trigger a full resync rather than simply trying to watch from the last event
-						// revision.
-						wc.logger.Debug("Watch was not closed by remote - full resync required")
-						wc.currentWatchRevision = ""
-						wc.onError()
-					}
-					wc.resyncAndCreateWatcher(ctx)
-				} else {
-					wc.onError()
-				}
+				// Handle a WatchError. This error triggered from upstream, all type
+				// of WatchError are treated equally,log the Error and trigger a full resync.
```

Comment:
This is a change of behavior isn't it?  Is it really necessary to resync every time we get an error?

Reply:
Yes. I was thinking to keep the error handling part lean. if Upstream reports a watch API error, log the error and act on that. I can bring back previous logic to wait until we reach the threshold limit.

Reply:
I think it's ok.  Just wondering if are simplifying then do we need the threshold error counts elsewhere in this module or whether we can just get rid of that completely?  WDYT?

Reply:
Removed threshold-based error handling in watcher cache.